apps/api/internal/auth/register.go

```
@@ -38,6 +38,45 @@ func (s *Service) RegisterSelfServe(
+	// Fails closed on an answer it cannot read: unreadable is not "unowned".
+	// Doing nothing costs a legitimate signup one retry; the alternative costs
+	// an operator their install.
+	if owned, oerr := s.repo.OwnershipEstablished(ctx); oerr != nil || !owned {
+		// The same address lookup the open path performs, discarded. Argon2
+		// above is the dominant cost and matching it closes most of the
+		// difference; this closes most of what remains, for one round trip and
+		// no behaviour. What cannot be matched without doing it is the write
+		// itself, so a residual difference stays — measured and reported rather
+		// than assumed away, in TestMeasureFirstRunRefusalTiming.
+		_, _ = s.repo.GetUserByEmail(ctx, in.Email)
+		return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 69) but it returns nil

(nilerr)